### PR TITLE
feat/ccn3-1012/health check

### PR DESCRIPTION
**Pull request by GeevesH-IS**
## Description
 
Pull request for overnight build, do not review currently

## Todo's before PR

- [x] Branch has been rebased against develop
- [x] Unit tests passed (`make test`)
- [x] PR named appropriately - see [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [x] PR description describes the overall goals of PR commits
- [x] Templates have been checked against the relevant user-story

## Migrations

- [x] Null fields in models specifies default value
- [x] You generated and applied migrations (`make migrate`)
- [x] You updated fixtures (`make export`)

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [ ] Specified reviewers (even if it's yourself)
- [x] Specified assignees (those who'll merge)
- [x] Specified labels

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [ ] Code syntax formatting checked
- [ ] Debug messages are absent

## PR merge

Select only one:

- [ ] Should be merged?
- [x] Should be rebased?
- [ ] Should be squashed?

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

**Comment by GeevesH-IS** [comment]
run jenkins build

**Comment by cruddasj** [comment]
@GeevesH-IS - checks now all pass (albeit the tests around household members have needed to be disabled due to change of workflow). Can you please ensure manual testing is performed over new workflow and once we're sure all is working as expected, merge down into develop then run into BUILD environment.

File: application/business_logic.py [hunk 1]
```diff
@@ -501,8 +502,14 @@ def remove_adult(application_id_local, remove_person):
     :param remove_person: adult to remove (integer)
     :return:
     """
+    application = Application.objects.get(pk=application_id_local)
+
     if AdultInHome.objects.filter(application_id=application_id_local, adult=remove_person).exists() is True:
         AdultInHome.objects.get(application_id=application_id_local, adult=remove_person).delete()
+        # Reset task status to WAITING if adults are updated
+        if application.people_in_home_status == 'WAITING':
+            application.people_in_home_status == 'IN_PROGRESS'
```
Comment (andrewskeaIS) [inline application/business_logic.py]: Change to single equal sign?

File: application/business_logic.py [hunk 2]
```diff
@@ -521,6 +530,10 @@ def rearrange_adults(number_of_adults, application_id_local):
                 next_adult_record = AdultInHome.objects.get(application_id=application_id_local, adult=next_adult)
                 next_adult_record.adult = i
                 next_adult_record.save()
+                # Reset task status to WAITING if adults are updated
+                if application.people_in_home_status == 'WAITING':
+                    application.people_in_home_status == 'IN_PROGRESS'
```
Comment (andrewskeaIS) [inline application/business_logic.py]: Change to single equal sign

File: application/tests/tests_selenium/test_apply_as_a_childminder.py [hunk 3]
```diff
@@ -319,39 +319,39 @@ def test_can_complete_people_in_your_home_task_without_other_adults_but_with_oth
...
-            faker.first_name(), faker.first_name(), faker.last_name_female(),
-            random.randint(1, 29), random.randint(1, 12), random.randint(1950, 1990), 'Friend', 121212121212,
-            False, None, None, None,
-            None, None, None,
-            None
-        )
-
-        self.assertEqual("Done", self.selenium_task_executor.get_driver().find_element_by_xpath(
-            "//tr[@id='other_people']/td/a/strong").text)
+    # @try_except_method
```
Comment (andrewskeaIS) [inline application/tests/tests_selenium/test_apply_as_a_childminder.py]: Should this be commented out?

File: application/tests/tests_selenium/test_apply_as_a_childminder.py [hunk 4]
```diff
@@ -559,7 +557,7 @@ def test_can_apply_as_a_childminder_using_full_question_set(self):
         """
         Test to exercise the largest set of questions possible within a childminder application
         """
-        self.create_standard_eyfs_application()
+        applicant_email = email = self.create_standard_eyfs_application()
```
Comment (andrewskeaIS) [inline application/tests/tests_selenium/test_apply_as_a_childminder.py]: The second email is not in use, can this be removed?

File: application/views/other_people_health_check/current_treatment.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    template_name = 'other_people_health_check/current_treatment.html'
+    form_class = CurrentIllness
+    success_url = 'Health-Check-Serious-Start'
+
+    def get_initial(self):
+        """
+        Get initial defines the initial data for the form instance that is to be rendered on the page
+        :return: a dictionary mapping form field names, to values of the correct type
+        """
+        initial = super(CurrentTreatment, self).get_initial()
```
Comment (andrewskeaIS) [inline application/views/other_people_health_check/current_treatment.py]: In Python 3 you can use super().class_function()

File: application/views/other_people_health_check/health_check_login.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+def has_expired(expiry):
+    """
+    Method to check whether a Magic Link URL has expired
+    :param expiry:
+    :return:
+    """
+    # Expiry period is set in hours in settings.py
+    exp_period = settings.EMAIL_EXPIRY * 60 * 60
+    diff = int(time.time() - expiry)
+    if diff < exp_period or diff == exp_period:
```
Comment (andrewskeaIS) [inline application/views/other_people_health_check/health_check_login.py]: Could use: diff <= exp_period

**Pull request merged**